tools/sl4a_client.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python3.4
+#
+# Copyright 2016 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tool to interactively call sl4a methods.
+
+Usage:
+$ sl4a_client
+Connected to 008d1d9749813804. Call sl4a methods against 'd' or 'droid'
+>>> droid.telephonyGetSimOperatorName()
+u'Fi Network'
+"""
+from __future__ import print_function
+
+import argparse
+import code
+import pprint
+import sys
+
+from mobly.controllers import android_device
+
+# Dictionary of variables available by default to the console shell.
+gConsoleEnv = {}
+
+
+def _StartConsole(banner=None):
+  code.interact(banner=banner, local=gConsoleEnv)
+
+
+def _Connect():
+  ad = gConsoleEnv['ad']
+  (droid, ed) = ad.get_droid()
+  gConsoleEnv['d'] = droid
+  gConsoleEnv['droid'] = droid
+  gConsoleEnv['ed'] = ed
+  gConsoleEnv['ed'].start()
+  print('Connection succeeded')
+
+
+def main(serial):
+  if not serial:
+    serials = android_device.list_adb_devices()
+    if len(serials) != 1:
+      raise Exception('Expected 1 phone, but %d found. Use the -s flag.' %
```

Comment:
can print error and exit(1) instead? cleaner console output that way since it's a cli tool.
similarly maybe the 'assert' below could be an if...exit check as well. Right now that 'assert' doesn't have any message, which could be confusing for cli tool users.